server/apps/immich/src/api-v1/asset/asset-repository.ts

```diff
@@ -174,7 +175,7 @@ export class AssetRepository implements IAssetRepository {
   async getDetectedObjectsByUserId(userId: string): Promise<CuratedObjectsResponseDto[]> {
     return await this.assetRepository.query(
       `
-        SELECT DISTINCT ON (unnest(si.objects)) a.id, unnest(si.objects) as "object", a."resizePath", a."deviceAssetId", a."deviceId"
+        SELECT DISTINCT ON (unnest(si.objects)) a.id, unnest(si.objects) as "object", a."resizePath", a."checksum", a."deviceId"
```

Comment:
I don't think this change is necessary, since we still have device asset id everywhere.

Reply:
Device asset id isn't null, but it will often be empty. Does this change the equation?